Move profiling docs to the Observability docs (#2785)
elastic/observability-docs: docs/en/observability/profiling-troubleshooting.asciidoc

+
![restart-integrations-server](./img/restart-integrations-server.png)
+
For non-demo workloads, verify that the Integrations Server has at least the recommended 4GB of RAM. You can check this on the Integrations Server page under *Instances*.

If you're unable to find a solution to the host-agent failure, you can raise a support request indicating `Universal Profiling` and `host-agent` as the source of the problem.

[discrete]
[[profiling-enable-verbose-logging]]
== Enable verbose logging in host-agent

During the support process, you may be asked to provide debug logs from one of the host-agent installations from your
deployment.

To enable debug logs, add the `-verbose` command-line flag or the `verbose true` setting in the configuration file.

NOTE: We recommend only enabling debug logs on a single instance of host-agent rather than an entire deployment to limit the amount of logs produced.

[discrete]
[[profiling-improve-load-time]]
== Improve load times

The amount of data loaded for the flamegraph, topN functions, and traces view can lead to latency when using a slow connection (e.g. DSL or mobile).

> If you're running Kibana/ES 8.6+, setting the Kibana cluster option `server.compression.brotli.enabled: true` reduces the amount of data transferred and should reduce load time.

Reviewer: Given that this guide will only be available starting with 8.7.0 we can remove the statement about the minimum version of 8.6+?

[[profiling-troubleshooting]]
= Troubleshoot your Universal Profiling 

++++
<titleabbrev>Troubleshoot</titleabbrev>
++++

beta::[]

You can use the host-agent logs to find errors.

The following is an example of a _healthy_ host-agent output:

> ----

Reviewer 2: All of the code blocks should specify a source: This differentiates a block a from a code block and will enable code highlighting (if applicable).

```suggestion
[source,logs]
----

```

[discrete]
[[profiling-troubleshoot-kubernetes]]
== Troubleshoot host-agent {k8s} deployments

When the Helm chart installation finishes, the output has instructions on how to check the host-agent pod status and read logs.
The following sections provide potential scenarios when host-agent installation *is not healthy*.

[discrete]
[[profiling-taints]]
=== Taints

{k8s} clusters often include https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/[taints and tolerations] in their setup.
In these cases, a host-agent installation may show no pods or very few pods running, even for a large cluster.

This is because a _taint_ precludes the execution of pods on a node unless the workload has been _tolerated_.
The Helm chart `tolerations` key in the `values.yaml` sets the toleration of taints using the official {k8s} scheduling API
format.

The following examples provide a `tolerations` config that you can add to the Helm chart `values.yaml`:

* To deploy the host-agent on all nodes with taint `workload=python:NoExecute`, add the following to the `values.yaml`:
+ 

> ----

Reviewer 2: Another good example. This one should be source,yaml

```suggestion
[source,yaml]
----
```

Reviewer 2: I'm not going to comment on all of them—you get the idea :)

time="..." level=info msg="Supports eBPF map batch operations"
time="..." level=info msg="eBPF tracer loaded"
time="..." level=info msg="Attached tracer program"
time="..." level=info msg="Attached sched monitor"
----

A host-agent deployment is working if the output of the following command is empty:

[source,logs]
----
head host-agent.log -n 15 | grep "level=error"
----

If running this command outputs error-level logs, the following are possible causes:

* The host-agent is running on an unsupported version of the Linux kernel, or its missing kernel features.
+
If the host-agent is running on an unsupported kernel version, the following is logged:
+
[source,logs]
----
Host Agent requires kernel version 4.15 or newer but got 3.10.0
----
+
If eBPF features are not available in the kernel, the host-agent fails to start, and one of the following is logged:
+
[source,logs]
----
Failed to probe eBPF syscall
----
+
or
+
[source,logs]
----
Failed to probe tracepoint
----
* The host-agent is not able to connect to {ecloud}. In this case, a similar message to the following is logged:
+
[source,logs]
----
Failed to setup gRPC connection (retrying...): context deadline exceeded
----
+

> Verify the `collection-agent` configuration value is set and is equal to what was printed by `elastic-profiling config`.

Reviewer: @inge4pres as we now have a single click setup in Kibana, this is outdated, right? Does it make sense to fix this in a separate PR?

Reviewer 3: Correct. I think we should amend it in this PR.
The instructions are printed directly from Kibana "Add Data", so we could say

```suggestion
Verify the `collection-agent` configuration value is set and is equal to what was printed in Kibana, when clicking to *Add Data*.
```

* The secret token is not valid, or it has been changed. In this case, the host-agent shuts down, and logs a similar message to the following:
+
[source,logs]
----
rpc error: code = Unauthenticated desc = authentication failed
----

> * The host-agent is unable to reach the `collectionagent` service. In this case, a similar message to the following is logged:

Reviewer: @inge4pres I don't think we ever explain in our docs what the "collection agent" is. Would it make sense to be a bit more vague here and instead write something along the lines of:

```
The host-agent is unable to send data to your deployment. In this case, a similar message to the following is logged:
```

Reviewer 3: I guess we were trying to reference the flag passed into the agent, which is still named `collection-agent` (also referenced above).
Your suggestion sounds good Daniel 👍🏼

Reviewer 3: I guess we were trying to reference the flag passed into the agent, which is still named `collection-agent` (also referenced above).
Your suggestion sounds good Daniel 👍🏼

+
[source,logs]
----
Failed to report hostinfo (retrying...): rpc error: code = Unimplemented desc = unknown service collectionagent.CollectionAgent"
----
+

> This typically means that your {ecloud} cluster has not been configured for Universal Profiling. To confingure your {ecloud} cluster, follow the steps in <<profiling-configure-data-ingestion,configure data ingestion>>.

Reviewer: Nit: `confingure` -> `configure`

The following are potential causes:

* {k8s} https://kubernetes.io/docs/concepts/services-networking/network-policies/[`NetworkPolicies`] define connectivity rules that prevent all outgoing traffic unless explicitly allow-listed.
* Cloud or datacenter provider network rules are restricting egress traffic to allowed destinations only (ACLs).

[discrete]
[[profiling-os-level-security]]
==== OS-level security

These settings _are not part of {k8s}_ and may have been included in the node setup. They can prevent the host-agent from working properly, as they intercept syscalls from the host-agent to the kernel and modify or block them.

If you have implemented security hardening (some providers listed below), you should know the privileges the host-agent needs.

* gVisor on GKE
* seccomp filters
* AppArmor LSM

[discrete]
[[profiling-submit-support]]
== Submit a support request

You can submit a support request from the https://cloud.elastic.co/support[support request page] in the {ecloud} console.

In the support request, specify if your issue deals with the host-agent or the Kibana app.

[discrete]
[[profiling-send-feedback]]
== Send feedback

> If troubleshooting and support are not fixing your issues, or you have any other feedback that you want to share about the

Reviewer: This is probably a question for @iogbole: Do we want to encourage people to send us an email to `profiling-feedback@elastic.co`? Usually we direct people to [our Discuss forum](https://discuss.elastic.co/) and I noticed there is already a [category for profiling](https://discuss.elastic.co/c/observability/profiling/92). Do we want to use this instead once the beta opens?

Reviewer 4: Good question, @danielmitterdorfer.  The email still applies in 8.7.  We recently created the profiling category in preparation for public beta.
